contracts/core/Registry.sol

```diff
@@ -1,47 +1,242 @@
+        );
+
+        return identityId;
+    }
+
+    /// @notice Updates the name of the identity and generates new anchor
+    /// @param _identityId The identityId of the identity
+    /// @param _name The new name of the identity
+    /// @dev Only owner can update the name.
+    function updateIdentityName(bytes32 _identityId, string memory _name)
+        external
+        isIdentityOwner(_identityId)
+        returns (address)
+    {
+        address anchor = _generateAnchor(_identityId, _name);
+
+        Identity storage identity = identitiesById[_identityId];
+        identity.name = _name;
+
+        // clear old anchor
+        anchorToIdentityId[identity.anchor] = bytes32(0);
+
+        // set new anchor
+        identity.anchor = anchor;
+        anchorToIdentityId[identity.anchor] = _identityId;
+
+        emit IdentityNameUpdated(_identityId, _name, anchor);
+
+        // TODO: should we return identity
+        return anchor;
+    }
+
+    /// @notice update the metadata of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _metadata The new metadata of the identity
+    /// @dev Only owner can update metadata
+    function updateIdentityMetadata(bytes32 _identityId, Metadata memory _metadata)
+        external
+        isIdentityOwner(_identityId)
+    {
+        identitiesById[_identityId].metadata = _metadata;
+
+        emit IdentityMetadataUpdated(_identityId, _metadata);
+    }
+
+    /// @notice Returns if the given address is an owner or member of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _account The address to check
+    function isOwnerOrMemberOfIdentity(bytes32 _identityId, address _account) public view returns (bool) {
+        return isOwnerOfIdentity(_identityId, _account) || isMemberOfIdentity(_identityId, _account);
+    }
+
+    /// @notice Returns if the given address is an owner of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _owner The address to check
+    function isOwnerOfIdentity(bytes32 _identityId, address _owner) public view returns (bool) {
+        return identitiesById[_identityId].owner == _owner;
+    }
+
+    /// @notice Returns if the given address is an member of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _member The address to check
+    function isMemberOfIdentity(bytes32 _identityId, address _member) public view returns (bool) {
+        return hasRole(_identityId, _member);
+    }
 
+    /// @notice Transfers the ownership of the identity to a new owner
+    /// @param _identityId The identityId of the identity
+    /// @param _owner New Owner
+    /// @dev Only owner can transfer ownership.
+    function changeIdentityOwner(bytes32 _identityId, address _owner) external isIdentityOwner(_identityId) {
+        identitiesById[_identityId].owner = _owner;
+        emit IdentityOwnerUpdated(_identityId, _owner);
     }
```

Comment:
We might want to make it a two step ownership transfer? Your call, just always scares me the risk of sending to wrong address. That would require adding a `pendingOwner` field as well and then having it be two calls.

Reply:
I like it